src/index.js

```diff
@@ -0,0 +1,96 @@
+const addTaskButton = document.getElementById('add-task-btn');
+const newTaskInput = document.getElementById('new-task-input');
+const todoList = document.getElementById('todo-list');
+const clearCompletedButton = document.getElementById('clear-completed-btn');
+
+function deleteTask(todoItem) { todoItem.remove(); }
+
+function editTask(todoItem) {
+  const taskElement = todoItem.querySelector('.task');
+  const taskText = taskElement.textContent;
+  taskElement.innerHTML = '';
+  const editInput = document.createElement('input');
+  editInput.type = 'text';
+  editInput.value = taskText;
+  editInput.classList.add('edit-input');
+  taskElement.appendChild(editInput);
+  editInput.focus();
+  editInput.addEventListener('keyup', (event) => {
+    if (event.key === 'Enter') {
+      const newText = editInput.value.trim();
+      if (newText !== '') {
+        taskElement.textContent = newText;
+        todoItem.style.backgroundColor = '';
+        todoItem.querySelector('.more-btn').style.display = 'inline-block';
+        todoItem.querySelector('.delete-btn').style.display = 'none';
+      }
+    }
+  });
+}
+
+function toggleDeleteButton(todoItem) {
+  const moreButton = todoItem.querySelector('.more-btn');
+  const deleteButton = todoItem.querySelector('.delete-btn');
+  moreButton.style.display = 'none';
+  deleteButton.style.display = 'inline-block';
+}
+
+function createTodoItem(taskText) {
```

Comment:
- [ ] You were expected to write a function to iterate over the tasks array and populate an HTML list item element for each task.
![image](https://github.com/Hsuwaddyhtat22/To-Do-List-mini-PJ/assets/62849872/052249db-97ab-43be-9dc3-50f0323b5c8a)
Based on your code, there is no tasks array explicitly defined or passed as an argument to any of the functions. However, there is a createTodoItem function that takes a taskText argument and returns an HTML li element that represents a single task.

It seems that the addTodoItem function is responsible for creating a new todo item (using the createTodoItem function) and appending it to the todoList element. This function is called when the user clicks the "Add Task" button, passing the value of the newTaskInput element as the taskText argument.

So while there is no explicit iteration over a tasks array, the addTodoItem function creates a new HTML element for each task added by the user. Kindly write the function as per project requirements